hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropPrimaryKeyTask.java

```diff
@@ -51,35 +54,51 @@ public DropPrimaryKeyTask(String theProductVersion, String theSchemaVersion, Str
 	private String generateSql() {
 		ourLog.debug("DropPrimaryKeyTask.generateSql()");
 
-		final ResultSetExtractor<String> resultSetExtractor = rs -> {
-			if (rs.next()) {
-				final String singleResult = rs.getString(1);
-
-				if (rs.next()) {
-					throw new IllegalArgumentException(Msg.code(2533)
-							+ "Expecting only a single result for the table primary but got multiple for task: "
-							+ getMigrationVersion());
-				}
-
-				return singleResult;
-			}
-			return null;
-		};
-
 		@Nullable
 		@Language("SQL")
 		final String primaryKeyNameSql = generatePrimaryKeyNameSql();
 
 		@Nullable
 		final String primaryKeyName = primaryKeyNameSql != null
-				? executeSqlWithResult(primaryKeyNameSql, resultSetExtractor, getTableNameWithDatabaseExpectedCase())
+				? executeSqlWithResult(primaryKeyNameSql, getTableNameWithDatabaseExpectedCase())
```

Comment:
problem: This is confused.  We DO want to run this query during a dry run - i.e. fetching the table name.  It is only the actual DROP command we don't want to run.  That's why we shouldn't be using methods named `execute*` for querying the metatdata.

Please just replace this with something simple like
```suggestion
				? newJdbcTemplate().queryForObject("some sql", String.clasS); 
```